tests/e2e-openshift-object-stores/aws-sts-tempostack/aws-sts-s3-create.sh

```diff
@@ -0,0 +1,67 @@
+#!/bin/bash
+set -euo pipefail
+
+# This script is meant to be run from OpenShift CI environment
+# Run the script with ./aws-sts-s3-create.sh TEMPO_NAME TEMPO_NAMESPACE
+
+# Set the AWS credential file var
+export AWS_SHARED_CREDENTIALS_FILE=$CLUSTER_PROFILE_DIR/.awscred
+export AWS_PAGER=""
+region=us-east-2
+tempo_name="$1"
+tempo_ns="$2"
+bucket_name=tracing-$tempo_ns-$OPENSHIFT_BUILD_NAMESPACE
+
+# Create a S3 bucket 
+aws s3api create-bucket --bucket $bucket_name --region $region --create-bucket-configuration LocationConstraint=$region
+
+# Set required vars to create AWS IAM policy and role
+oidc_provider=$(oc get authentication cluster -o json | jq -r '.spec.serviceAccountIssuer' | sed 's~http[s]*://~~g')
+aws_account_id=$(aws sts get-caller-identity --query 'Account' --output text)
+cluster_id=$(oc get clusterversion -o jsonpath='{.items[].spec.clusterID}{"\n"}')
+trust_rel_file="/tmp/$tempo_ns-trust.json"
+role_name="tracing-$tempo_ns-$OPENSHIFT_BUILD_NAMESPACE"
+
+# Create a trust relationship file
+cat > "$trust_rel_file" <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Principal": {
+        "Federated": "arn:aws:iam::${aws_account_id}:oidc-provider/${oidc_provider}"
+      },
+      "Action": "sts:AssumeRoleWithWebIdentity",
+      "Condition": {
+        "StringEquals": {
+          "${oidc_provider}:sub": [
+            "system:serviceaccount:${tempo_ns}:tempo-${tempo_name}",
+            "system:serviceaccount:${tempo_ns}:tempo-${tempo_name}-query-frontend"
```

Comment:
Is this line required? There is no `tempo-${tempo_name}-query-frontend` service account, neither in TempoStack nor in TempoMonolithic.

Reply:
@andreasgerstmayr The SA is created when we enable the ingress type as route in TempoStack instance. However its not created for the Monolithic, which uses a single SA.

Reply:
https://github.com/grafana/tempo-operator/pull/978#issuecomment-2224674870

Reply:
Oh, sorry I didn't realize we're creating a new SA here.